### PR TITLE
[Review] Plugin handling of Ethernet and EthernetETF

### PR DESCRIPTION
**Pull request by opcua-tsn-team-kalycito**
- Removed the separate usage of ETF plugin

- Removed the hardcoding of subscribing interval and socket timeout by creating setDefaultConfing API of ReaderGroup

### CODE REVIEW

File: examples/pubsub_realtime/README_pubsub_TSN.txt [hunk 1]
```diff
@@ -127,7 +127,7 @@ If MAC address is changed, follow INGRESS POLICY steps with dst_mac address to b
 
 To write the published counter data along with timestamp in csv uncomment #define UPDATE_MEASUREMENTS in pubsub_TSN_publisher.c and pubsub_TSN_loopback.c applications
 
-NOTE: The CSV files will be generated in the build directory containing the timestamp and counter values for benchmarking only after terminating both the applications using �Ctrl + C�
+NOTE: The CSV files will be generated in the build directory containing the timestamp and counter values for benchmarking only after terminating both the applications using �Ctrl + C�
```
Comment (ccvca) [inline examples/pubsub_realtime/README_pubsub_TSN.txt:130]: Is this intended?

File: plugins/ua_pubsub_ethernet.c [hunk 2]
```diff
@@ -290,6 +387,69 @@ UA_PubSubChannelEthernet_unregist(UA_PubSubChannel *channel,
...
+    message.msg_control    = dataPacket;
+    /* Provide the size of necessary bytes */
+    message.msg_controllen = sizeof(dataPacket);
+    /* Structure for storing the necessary data */
+    struct cmsghdr*       controlMsg;
+    /* Control message created for tx time */
+    controlMsg             = CMSG_FIRSTHDR(&message);
+    controlMsg->cmsg_level = SOL_SOCKET;
+    controlMsg->cmsg_type  = SCM_TXTIME;
+    controlMsg->cmsg_len   = CMSG_LEN(sizeof(__u64));
```
Comment (ccvca) [inline plugins/ua_pubsub_ethernet.c:440]: This should be 'UA_UInt64' as well

File: plugins/ua_pubsub_ethernet.c [hunk 3]
```diff
@@ -290,6 +387,69 @@ UA_PubSubChannelEthernet_unregist(UA_PubSubChannel *channel,
...
+    if(ethernettransportSettings && (ethernettransportSettings->transmission_time != 0))
+        *((__u64 *) CMSG_DATA(controlMsg)) = ethernettransportSettings->transmission_time;
```
Comment (ccvca) [inline plugins/ua_pubsub_ethernet.c:442]: This should be 'UA_UInt64' as well

**Review by ccvca — CHANGES_REQUESTED** [review]
There are also several permission changes from 0644 to 0755 for .c and .h files. The old permissions seems to be right, as these files are not executable.

File: examples/pubsub_realtime/README_pubsub_TSN.txt (hunk 1 above)
Comment (ccvca) [reply examples/pubsub_realtime/README_pubsub_TSN.txt:130]: grafik
Might be a bug on the file view.

File: plugins/ua_pubsub_ethernet.c (hunk 3 above)
Comment (opcua-tsn-team-kalycito) [reply plugins/ua_pubsub_ethernet.c]: Thanks, done.

**Comment by opcua-tsn-team-kalycito** [comment]
@jpfr could you re-trigger the failed Windows CI build? The error seems to be in the client_subscription test case which is not related to this PR.

**Pull request merged**